### PR TITLE
deploy doxygen document

### LINKED ISSUE #8: GitHub Pages
Make documentation in GitHub pages

### PR DESCRIPTION
**Pull request by Quesys-tech**
Closes #8

### CODE REVIEW

File: docs/mainpage.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Basic Code Documentation {#mainpage}
```
Comment (zukain5) [inline docs/mainpage.md:1]:
```suggestion
# MPS-Basic Documentation {#mainpage}
```

File: docs/mainpage.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+Modernized Moving Particle Semi-implicit/Simulation code written in C++.
+
+## Coding guide
+- C++ standard: C++17
+- Do **NOT** use Exceptions
+- Macros
+	- Defined in `common.hpp`
+    - Excessive use should be avoided
+	- You can use `rep(i, a, b)` if you need to write `for (int i = a; i < b; i++)`
+		- However, **use range-based for `for (auto& e : v)` instead whenever it is possible**
```
Comment (zukain5) [inline docs/mainpage.md:12]: I think coding guide is not needed in the main page

**Review by zukain5 — CHANGES_REQUESTED** [review]
If we use the Doxygen documentation as the main documentation for your project, you may want to reduce the README to avoid duplication.

**Review by iori3150 — APPROVED** [review]
Thank you!

**Review by iori3150 — APPROVED** [review]

**Comment by iori3150** [comment]
とりあえず見たかったのでマージしちゃいました笑

**Comment by zukain5** [comment]
まあ大丈夫でしょう！w

**Pull request merged**